analytics/pubstack/pubstack_module.go

```diff
@@ -0,0 +1,264 @@
+	if err != nil {
+		glog.Error(err)
+		return nil, err
+	}
+	go pb.start(configUrl, refreshDelay)
+	go func() { _ = pb.reloadConfig(configUrl) }()
+
+	glog.Info("[pubstack] Pubstack analytics configured and ready")
+	return &pb, nil
+}
+
+func (p *PubstackModule) LogAuctionObject(ao *analytics.AuctionObject) {
+	p.muxConfig.RLock()
+	defer p.muxConfig.RUnlock()
+
+	if !p.isFeatureEnable(auction) {
+		return
+	}
+
+	// serialize event
+	payload, err := helpers.JsonifyAuctionObject(ao, p.scope)
+	if err != nil {
+		glog.Warning("[pubstack] Cannot serialize auction")
+		return
+	}
+
+	p.eventChannels[auction].Push(payload)
+}
+
+func (p *PubstackModule) LogVideoObject(vo *analytics.VideoObject) {
+	p.muxConfig.RLock()
+	defer p.muxConfig.RUnlock()
+
+	if !p.isFeatureEnable(video) {
+		return
+	}
+
+	// serialize event
+	payload, err := helpers.JsonifyVideoObject(vo, p.scope)
+	if err != nil {
+		glog.Warning("[pubstack] Cannot serialize video")
+		return
+	}
+
+	p.eventChannels[video].Push(payload)
+}
+
+func (p *PubstackModule) LogSetUIDObject(so *analytics.SetUIDObject) {
+	p.muxConfig.RLock()
+	defer p.muxConfig.RUnlock()
+
+	if !p.isFeatureEnable(setUID) {
+		return
+	}
+
+	// serialize event
+	payload, err := helpers.JsonifySetUIDObject(so, p.scope)
+	if err != nil {
+		glog.Warning("[pubstack] Cannot serialize video")
+		return
+	}
+
+	p.eventChannels[setUID].Push(payload)
+}
+
+func (p *PubstackModule) LogCookieSyncObject(cso *analytics.CookieSyncObject) {
+	p.muxConfig.RLock()
+	defer p.muxConfig.RUnlock()
+
+	if !p.isFeatureEnable(cookieSync) {
+		return
+	}
+
+	// serialize event
+	payload, err := helpers.JsonifyCookieSync(cso, p.scope)
+	if err != nil {
+		glog.Warning("[pubstack] Cannot serialize video")
+		return
+	}
+
+	p.eventChannels[cookieSync].Push(payload)
+
+}
+
+func (p *PubstackModule) LogAmpObject(ao *analytics.AmpObject) {
+	p.muxConfig.RLock()
+	defer p.muxConfig.RUnlock()
+
+	if !p.isFeatureEnable(amp) {
+		return
+	}
+
+	// serialize event
+	payload, err := helpers.JsonifyAmpObject(ao, p.scope)
+	if err != nil {
+		glog.Warning("[pubstack] Cannot serialize video")
+		return
+	}
+
+	p.eventChannels[amp].Push(payload)
+
+}
+
+func (p *PubstackModule) reloadConfig(configUrl *url.URL) error {
+	config, err := fetchConfig(p.httpClient, configUrl)
+	if err != nil {
+		return err
+	}
+	p.configCh <- config
+	return nil
+}
+
+func (p *PubstackModule) start(configUrl *url.URL, refreshDelay time.Duration) {
+
+	tick := time.NewTicker(refreshDelay)
+
+	for {
+		select {
+		case <-p.sigTermCh:
+			p.closeAllEventChannels()
+			return
+		case config := <-p.configCh:
+			p.updateConfig(config)
+			glog.Infof("[pubstack] Updating config: %v", p.cfg)
+		case <-tick.C:
+			go func() {
+				err := p.reloadConfig(configUrl)
+				if err != nil {
+					glog.Errorf("[pubstack] Fail to fetch remote configuration: %v", err)
+				}
+			}()
+		}
+	}
+
+}
+
+func (p *PubstackModule) updateConfig(config *Configuration) {
```

Comment:
This could be great

But I try to respect the mantra "the best code is no code at all" :).

Because the update process is very infrequent (every 2h), I think the optimization will be very limited here. 
But because it's even more infrequent to have a configuration update, I can add a simple test to check if the config is new and if yes, update all channels

What do you think?

Reply:
> I can add a simple test to check if the config is new and if yes, update all channels

Sounds good to me.